app/services/setting_service.rb

```diff
@@ -0,0 +1,73 @@
+require_relative "data_sources/json_setting_data_source"
+require_relative "../shared/logging_module"
+require_relative "../shared/available_settings"
+require_relative "./services"
+
+module FastlaneCI
+  class SettingServiceKeyNotFoundError < FastlaneCIError
+  end
+
+  # Provides access to system settings
+  class SettingService
+    include FastlaneCI::Logging
+    attr_accessor :setting_data_source
+
+    def initialize(setting_data_source: nil)
+      unless setting_data_source.nil?
+        unless setting_data_source.class <= SettingDataSource
```

Comment:
nit: I think we should use `kind_of?` instead of this comparison operator. No need to change, we should update rubocop instead.

Reply:
I'm all for kind_of? I use it usually. In this case the code is copied from all the other services we already use to be consistent.